arex-schedule-web-api/src/main/java/com/arextest/schedule/planexecution/impl/PlanExecutionMonitorImpl.java

```diff
@@ -34,52 +32,56 @@ public void monitorOne(ReplayPlan task) {
     for (PlanMonitorHandler handler : planMonitorHandlerList) {
       try {
         handler.handle(task);
-      } catch (Throwable t) {
-        LOGGER.error("Error handling plan:{}", task.getId(), t);
+      } catch (Exception e) {
+        LOGGER.error("Error handling plan:{}", task.getId(), e);
       }
     }
   }
 
   @Override
   public void register(ReplayPlan plan) {
-    MonitorTask task = new MonitorTask(plan);
-    ScheduledFuture<?> monitorFuture = monitorScheduler
-        .scheduleAtFixedRate(task, 0, SECOND_TO_REFRESH, TimeUnit.SECONDS);
-    plan.setMonitorFuture(monitorFuture);
+    LOGGER.info("register monitor task {}", plan.getId());
+    for (PlanMonitorHandler handler : planMonitorHandlerList) {
+      monitorScheduler.scheduleAtFixedRate(new MonitorTask(plan, handler), 0, handler.period(), TimeUnit.SECONDS);
+    }
+    plan.setMonitorScheduler(monitorScheduler);
   }
 
   @Override
   public void deregister(ReplayPlan plan) {
-    plan.getMonitorFuture().cancel(false);
     LOGGER.info("deregister monitor task, planId: {}", plan.getId());
+    ScheduledExecutorService monitorExecutor = plan.getMonitorScheduler();
+    monitorExecutor.shutdown();
```

Comment:
How do other replay plan tasks run after shutdown?

Reply:
This is shutdown in the plan dimension and will not affect other replay plans.

Reply:
Each replay plan monitor task share the same thread pool: `monitorScheduler`, if this executor shutdown, the other monitor tasks will stop executing as well